### PR TITLE
8.0b3 installation changes

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Update installation instructions.    
* Document Conda Forge package outputs.        
* Pin documented Conda environments to the Cylc "release" (i.e. minor        
  version).        
* Update test workflow to create the new environments.        
* Add environment (and with it a test) for mixed Conda-Forge / Anaconda        
  installations.     

Note: the conda tests are expected to fail at this stage, see the comment below and a link to a run where they pass.

### CODE REVIEW

File: src/reference/environments/envs/cylc-all.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: cylc-|release|
```
Comment (oliver-sanders) [inline src/reference/environments/envs/cylc-all.yml:1]: This gets replaced with the release when the file is read into the docs so we don't have to update it.

**Comment by oliver-sanders** [comment]
Note to self: for the conda builds we need the cylc version which we get from cylc so we need cylc?!

**Comment by oliver-sanders** [comment]
The "create conda environments" tests will fail at the moment because they are trying to create a Cylc 8.0rc1 environment (because Cylc Flow master is 8.0rc1).

These tests will be a little flaky since they are trying to build environments for versions which have not necessarily been released yet. These tests will run whenever these files are changed, on release, or on request (can manually set the cylc/python versions for the latter).

File: .github/actions/create-conda-envs/action.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        # use this when released: https://github.com/mamba-org/setup-mamba
+        conda install -n base -c conda-forge -y mamba
+
+    - shell: bash
+      run: |
+        for env_file in $(echo src/reference/environments/envs/*.yml); do
+        echo "# creating env from: ${env_file}"
+          new_env_file="$(sed 's/\(.*\)\.yml/\1-2.yml/' <<< "${env_file}")"
+          # substitute the python and cylc-flow versions in the env files
+          # TODO: add explicit python dep
```
Comment (oliver-sanders) [inline .github/actions/create-conda-envs/action.yml:39]: For another day.

Comment (oliver-sanders) [reply .github/actions/create-conda-envs/action.yml:39]: https://github.com/cylc/cylc-doc/issues/343

File: src/installation.rst [hunk 3]
```diff
@@ -3,64 +3,53 @@
...
 
-   This documentation has been revised for Cylc 8, but some new features are
-   yet to be fully described.
+   Please report any issues to the `Cylc 8 Discourse Support Group`_.
 
 
 Quick Installation
 ------------------
 
 Cylc runs on Unix systems including Linux and Mac OS.
```
Comment (wxtim) [inline src/installation.rst:16]: Mostly irrelevant.
```suggestion
Cylc runs on Unix-like systems including Linux and Mac OS.
```

File: src/installation.rst [hunk 4]
```diff
@@ -3,64 +3,53 @@
...
-
-   * - .. rubric:: Via Conda (recommended):
-     - .. rubric:: Via Pip (+npm):
-
-   * - ::
+Via Conda (recommended)
+^^^^^^^^^^^^^^^^^^^^^^^
 
-          $ conda install cylc-flow
+.. code-block:: sub
```
Comment (wxtim) [inline src/installation.rst:21]: What is this lexer?

File: src/installation.rst [hunk 5]
```diff
@@ -119,16 +108,21 @@ your ``$PATH``, follow the instructions in the ``brew install`` output.
...
+For distributed and multi-user installation we recommend using Conda and
+installing Cylc components only where required.
 
-For multi-user installation we recommend using Conda and installing
-Cylc components only where required.
+.. tip::
 
-The Cylc Packages
-^^^^^^^^^^^^^^^^^
+   For example Conda environments and installation options see
```
Comment (wxtim) [inline src/installation.rst:119]: IMO slight change is easier to read.
```suggestion
   For examples of Conda environments and installation options see
```

**Review by wxtim — COMMENTED** [review]
- [x] Read - makes sense - I've put a couple of extremely minor suggestions down, nothing that wouldn't stop me approving.
- [x] Built locally.
- [x] Read built version.
- [x] Would like to know if test failing because of absence of rc1 is deliberate or a mistake. -
      ```
      Encountered problems while solving:
        - nothing provides requested cylc-flow 8.0rc1**
      ```

File: src/installation.rst (hunk 4 above)
Comment (oliver-sanders) [reply src/installation.rst:21]: https://cylc.github.io/cylc-sphinx-extensions/extensions/cylc.sphinx_ext.sub_lang.html

(simple lexer that is less likely to produce unexpected highlighting)

**Comment by oliver-sanders** [comment]
> Would like to know if test failing because of absence of rc1 is deliberate or a mistake.

Explained by my comments above.

**Review by wxtim — APPROVED** [review]

File: src/hyperlinks.rst.include [hunk 6]
```diff
@@ -6,7 +6,9 @@
 .. _Cylc User Guide: https://cylc.github.io/cylc-doc/latest/html/index.html
 .. _Cylc Workflow Design Guide: https://cylc.github.io/cylc-doc/latest/html/workflow-design-guide/index.html
 .. _Cylc: https://cylc.github.io/
+.. _Cylc 8 Discourse Support Group: https://cylc.discourse.group/c/cylc/7-to-8/13
```
Comment (datamel) [inline src/hyperlinks.rst.include:9]: Hmm that 13 is strange, although works nicely, just looks a bit odd.

**Review by datamel — APPROVED** [review]
No problems found by me.

File: src/hyperlinks.rst.include (hunk 6 above)
Comment (oliver-sanders) [reply src/hyperlinks.rst.include:9]: Discourse topic number?

**Pull request merged**